src/api/js/src/high-level/types.ts

```diff
@@ -0,0 +1,520 @@
+import {
+  Z3_ast,
+  Z3_ast_map,
+  Z3_ast_vector,
+  Z3_context,
+  Z3_decl_kind,
+  Z3_func_decl,
+  Z3_func_interp,
+  Z3_model,
+  Z3_pattern,
+  Z3_probe,
+  Z3_solver,
+  Z3_sort,
+  Z3_sort_kind,
+  Z3_symbol,
+  Z3_tactic,
+} from '../low-level';
+
+export type AnySort<Name extends string = any> =
+  | SortRef<Name>
+  | BoolSortRef<Name>
+  | ArithSortRef<Name>
+  | BitVecSortRef<Name>;
+export type AnyExpr<Name extends string = any> =
+  | ExprRef<Name>
+  | BoolRef<Name>
+  | PatternRef<Name>
+  | QuantifierRef<Name>
+  | ArithRef<Name>
+  | IntNumRef<Name>
+  | RatNumRef<Name>
+  | BitVecRef<Name>;
+export type AnyAst<Name extends string = any> = AnyExpr<Name> | AnySort<Name> | AstRef<Name> | FuncDeclRef<Name>;
```

Comment:
Not sure how AstRef plays in here as it should be one of Sort, Expr, FuncDecl.
We have a very few functions that work polymorphically.